File: java/L600/LeetCode0690.java
package L600;

import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

class Employee {
    public int id;
    public int importance;
    public List<Integer> subordinates;
}

public class LeetCode0690 {
    public int getImportance(List<Employee> employees, int id) {
        HashMap<Integer, Employee> map = new HashMap<>(employees.size());
        for (Employee e : employees) {
            map.put(e.id, e);
        }
        if (!map.containsKey(id)) {
            return 0;
        }
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(id);
        int ans = 0;
        while (!queue.isEmpty()) {
            int idt = queue.poll();
            Employee e = map.get(idt);
            ans += e.importance;
            for (Integer i : e.subordinates) {
                queue.offer(i);
            }
        }
        return ans;
    }
}

File: cpp/L600/LeetCode0690.cc
#include <unordered_map>
#include <vector>
using std::vector;

class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};

class Solution {
    int getImportance(int id, std::unordered_map<int, int> const &mp, vector<Employee*> const &employees) {
        auto it = mp.find(id);
        if (it == mp.end()) {
            return 0;
        }
        Employee *cur = employees[it->second];
        int importance = cur->importance;
        for (int nextId : cur->subordinates) {
            importance += getImportance(nextId, mp, employees);
        }
        return importance;
    }
public:
    int getImportance(vector<Employee*> employees, int id) {
        std::unordered_map<int, int> emplyeeMap;
        for (int i = 0;i < employees.size();++i) {
            emplyeeMap[employees[i]->id] = i;
        }
        return getImportance(id, emplyeeMap, employees);
    }
};